### PR TITLE
List builders

### PR DESCRIPTION
**Pull request by zapaz**
## Description

List current members of batch 5. 

Read Builders addresses from the EventLogs

## Related Issues

_Closes #3_

Your ENS/address: zapaz.eth

### CODE REVIEW

**Comment by derrekcoleman** [comment]
Glad to see the functionality implemented, @zapaz. Now let's give the design and layout some love. How could you arrange this information so it's easy for a user to see? Would a dense table or modern cards make more sense? [Here are some DaisyUI components](https://daisyui.com/components/) so you don't have to reinvent the wheel.

File: packages/nextjs/app/builders/page.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const contractLogs = useContractLogs(batchRegistryAddress);
+
+  useEffect(() => {
+    const logBuilders = contractLogs
+      .map(log => decodeEventLog({ abi: batchRegistryAbi, data: log.data, topics: log.topics }))
+      .filter(log => log.eventName === "CheckedIn")
+      .map(log => (log.args as { builder: string }).builder);
+
+    setBuilders(logBuilders);
+  }, [contractLogs]);
```
Comment (technophile-04) [inline packages/nextjs/app/builders/page.tsx:64]: We have nice [`useScaffoldEvenHIstory`](https://docs.scaffoldeth.io/hooks/useScaffoldEventHistory) hook which can simplify this

Comment (zapaz) [reply packages/nextjs/app/builders/page.tsx:64]: ok!

Comment (zapaz) [reply packages/nextjs/app/builders/page.tsx]: done in last commit #c2df15

File: packages/nextjs/app/builders/page.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        <div className="bg-base-100 border-base-300 border shadow-md shadow-secondary rounded-3xl px-6 lg:px-8 mb-6 space-y-1 py-4 overflow-x-auto">
+          <Link href={`/builders/${address}`}>
+            {isFetchedAvatar ? (
+              <BlockieAvatar address={address} ensImage={ensAvatar} size={100} />
+            ) : (
+              <span className="loading loading-spinner loading-lg"></span>
+            )}
+            <div>{isFetchedEns ? ensName : "*****"}</div>
+            <div>{address}</div>
+          </Link>
```
[the remark this answers is not in the document]
Comment (zapaz) [reply packages/nextjs/app/builders/page.tsx]: `<Address />` component used

**Comment by zapaz** [comment]
> Here are some DaisyUI components

[Daisy UI timeline component](https://daisyui.com/components/timeline/) used

File: packages/nextjs/app/builders/page.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  return (
+    <>
+      <ul className="timeline timeline-vertical">
+        <li>
+          <div className="timeline-middle timeline-box">List of Batch #5 Builders</div>
+          <hr />
+        </li>
+        {data && data.length > 0 ? (
+          data.map((log, index) => <Builder key={index} address={log.args.builder || ""} />)
```
Comment (technophile-04) [inline packages/nextjs/app/builders/page.tsx:42]: Same builder can checkin multiple times right ? So there is possibility that we get duplicate addresses when seeing through all "CheckedIn" events, 

maybe do : 

```ts
 const builderAddresses = data?.map(event => event.args.builder as string | undefined)
 const uniqueBuilders = Array.from(new Set(builderAddresses));
```

And then map over uniqueBuilders while rendering component ?

Comment (zapaz) [reply packages/nextjs/app/builders/page.tsx:42]: yes, I re-check and appeared twice...

Comment (zapaz) [reply packages/nextjs/app/builders/page.tsx]: fixed in last commit

**Review by technophile-04 — APPROVED** [review]

**Pull request merged**